vignettes/icommands.Rmd

```diff
@@ -0,0 +1,203 @@
+---
+title: "rirods vs iCommands"
+author: "Mariana Montes"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{rirods vs iCommands}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+library(rirods)
+# httptest2::start_vignette("icommands")
+use_irods_demo("rods", "rods")
+create_irods("http://localhost/irods-rest/0.9.3", "/tempZone/home", overwrite=TRUE)
+iauth("rods", "rods")
+if (length(ils())> 0) {
+  for (file in ils()$logical_path) {
+    irm(file, recursive = TRUE)
+  }
+}
+imkdir("some_collection")
+x <- rnorm(100)
+isaveRDS(x, "some_collection/random_numbers.rds")
+imkdir("some_collection/subcollection")
+imeta("some_collection/random_numbers.rds",
+      operations = list(
+        list(operation="add", attribute="length", value="100", units="items"),
+        list(operation="add", attribute="distribution", value="normal")
+      ))
+y <- rnorm(200)
+isaveRDS(y, "200numbers.rds")
+
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval=is_irods_demo_running()
+)
+
+knitr::opts_knit$set(
+  root.dir = tempdir()
+)
+
+```
+
+This article is meant for iRODS users familiar with iCommands, the command-line interface to iRODS. While `{rirods}` functions are based on iCommands, not all iCommands are covered and some functionality may differ. The article also compares the bash calls to the R calls in case the reader is not so familiar with R syntax.
+
+The table below lists the main functions available in `{rirods}` and their counterparts in iCommands.
+
+`{rirods}` | `iCommands`
+----- | -----
+`icd()` | `icd`
+`ipwd()` | `ipwd`
+`ils()` | `ils`
+`iget()`, `ireadRDS()` | `iget`
+`iput()`, `isaveRDS()` | `iput`
+`imeta()` | `imeta`
+`iquery()` | `iquest`
+`imkdir()` | `imkdir`
+`irm()` | `irm`
+
+A general note is that in iCommands the `-h` argument provides the documentation for a certain command (e.g. `icd -h`), whereas in R this is achieved by preceding the name of the function with `?` (e.g. `?icd`). Moreover, the `verbose` boolean argument in most of `{rirods}` commands is used to print information about the HTTP request and response (and is by default `FALSE`), which is not a behaviour relevant to iCommands.
+
+## Working directory and collections
+
+The iCommand `ipwd` and its `{rirods}` counterpart `ipwd()` work exactly the same, providing the path to the current working directory. The case of `icd()` is different; unlike `icd`, which returns the "/zone/home" directory if no argument is provided, `icd()` needs an argument with the path for a new working directory. `icd("/tempZone/home")` will take us to `"/tempZone/home"` like `icd` might, but `icd()` with no arguments will throw an error.
```

Comment:
Oh, is that true?  `icd()` with no arguments won't go back to the 'home collection'?    Is that on purpose?    Is there a downside to having that functionality be implemented?

Reply:
@MartinSchobben ?
I remember that we had an issue to try to bring `icd()` closer to the behavior of `icd` but I don't remember right now why this part was not implemented (whether it was just missed or on purpose).
The local-R counterpart `setwd()` always requires an argument because it doesn't have the notion of a home directory, so maybe the idea was to keep it in line with that?

Reply:
Hmm, that does sound vaguely familiar.    An interesting case when Linux and R diverge...

Reply:
I think it was indeed that I modelled this after R. Maybe we should check this once and see what we prioritise, mimic r or irods behaviour.